packages/react-native/src/hooks.tsx

```diff
@@ -41,11 +42,31 @@ export function useIsStorySelected(storyId: string) {
 }
 
 /**
- * Hook that indicates if `title` is the currently selected story section.
+ * Hook that indicates if story kind (title) is the currently selected story section.
  */
 export function useIsStorySectionSelected(title: string) {
   return useAtomValue(
-    useMemo(() => atom((get) => get(storyContextAtom)?.title === title), [title])
+    useMemo(
+      () =>
+        atom((get) => {
+          const contextTitle = get(storyContextAtom)?.title;
+          return contextTitle === title || contextTitle?.startsWith(`${title}/`);
+        }),
+      [title]
+    )
+  );
+}
+
+export function useIsChildSelected(entries: StoryIndexEntry[]) {
+  return useAtomValue(
+    useMemo(
+      () =>
+        atom((get) => {
+          const contextId = get(storyContextAtom)?.id;
+          return !!entries.find(({ id }) => id === contextId);
+        }),
+      [entries]
+    )
```

Comment:
It's not super clear from reading the calling code whether `item.stories` is a stable reference. Does it only change when Storybook core needs to update (because of HMR or whatever else)?

Reply:
When filtering the tree can be recreated so I guess not stable in some sense.